std/experimental/logger/asynclogger.d

```diff
@@ -0,0 +1,161 @@
+module std.experimental.logger.asynclogger;
+
+import std.experimental.logger.core;
+import std.concurrency;
+
+/** The $(D AsyncLogger) processes logs asynchronously by using another $(D Logger) implementation
+as a backend. Every data logged to this $(D AsyncLogger) will be send to separate logging thread
+in order to be logged there. It can help to improve logging performance for $(D log) caller,
+for example by removing waiting for flushing logs to a file.
+Example:
+-------------
+void main()
+{
+    import std.algorithm;
+    import std.conv;
+    import std.stdio;
+    import std.datetime;
+    import std.experimental.logger;
+
+    auto fl = new FileLogger("defaultFileLogger.log", LogLevel.all);
+    void logFileDefault()
+    {
+        fl.log("test message");
+    }
+
+    auto afl = new AsyncLogger(new FileLogger("asyncFileLogger.log", LogLevel.all));
+    void logFileAsync()
+    {
+        afl.log("test message");
+    }
+
+    auto r = benchmark!(logFileDefault, logFileAsync)(100_000);
+    writeln(map!(a => to!string(to!Duration(a)))(r[]));
+}
+-------------
+*/
```

Comment:
This unittest requires TestLogger class that is inaccessible from user code. May be it is better to add simple unittest based on NullLogger, just for example. Or no example at all?